src/pacomo.js

```diff
@@ -107,11 +107,20 @@ export function transformWithPrefix(strategy, componentFunction) {
       typeof element.type !== 'function'
     )
 
-    if (element.props.className) {
-      changes.className = `${rootClass || ''} ${prefixedClassNames(strategy, componentFunction, element.props.className)} ${suffixClasses}`
+    // make any necessary class name updates
+    let newClassNames = [];
+    if(rootClass && rootClass.length > 0) {
+        newClassNames.push(rootClass);
```

Comment:
you should be able to use the `classnames` library to do all of this and greatly simplify it.
```
  changes.className = classnames(rootClass, 
    prefixedClassNames(strategy, componentFunction, element.props.className),
    suffixClasses
);
```